before_after_filesystem_snapshot.py

```diff
@@ -0,0 +1,65 @@
+def snapshot(before_dict, after_dict):
+  '''before_after_snapshot is a simple function that returns which files were
+    unchanged, modified, added or removed from an input dictionary (before_dict)
+    and an output dictionary (after_dict). Both these dictionaries have file
+    names as the keys and their hashes as the values.'''
+
+  unchanged_files = []
+  modified_files = []
+  added_files = []
+  removed_files = []
+  for key in before_dict:
+    if key in after_dict:
+      if before_dict[key] == after_dict[key]:
+        # Matching the hashes to check if file was unchanged
+        unchanged_files.append(key)
+      else:
+        modified_files.append(key)
+    else:
+      removed_files.append(key)
+  for key in after_dict:
+    if key not in before_dict:
+      # Looking for new files
+      added_files.append(key)
+
+  # Returning the snapshot of the new file system
+  return (sorted(unchanged_files), sorted(modified_files), sorted(added_files),
+  sorted(removed_files))
+
+def generate_artifact_rules(snapshot):
+  '''
+  Generate Artifact Rules given which files have been added, which have been removed,
+  which have been modified, and which have remained unchanged. 
+  '''
+  expected_materials = []
+  expected_products = []
+
+  # missing rules for MATCH since we don't have the information of the material from 
```

Comment:
Let's start this comment with `# TODO: ` and take care of it in a follow up PR.

Reply:
Thank you for so many reviews Lukas! I added `#TODO: ` and the Travis check runs and passes. I think it is ready to merge now. 

> Could you please also create a new issue that very briefly and roughly describes what we want to do next (read in link metadata, generate match rules that span multiple links, etc...)?
Sure.